x-pack/plugins/security_solution/public/common/components/header_actions/actions.tsx

```diff
@@ -81,6 +86,17 @@ const ActionsComponent: React.FC<ActionProps> = ({
 
   const isEnterprisePlus = useLicense().isEnterprise();
 
+  const panels = useExpandableFlyoutState();
```

Comment:
can't this just be done in the GuidedOnboardingTourStep? This doesn't need to be a prop passed in, can be derived from this hook. We should also only render 1 GuidedOnboardingTourStep, change line 277 to `{showExpandEvent && isTourAnchor && (`

Reply:
This would prevent re-renders in any component in the table except for the active tour. If we pushed the state down even further into SecurityTourStep I think re-rendering would only occur when the tour is active, and so this pr would introduce basically 0 new re-renders

Reply:
That makes sense. I tried to access `useExpandableFlyoutState` from GuidedOnboardingTourStep `x-pack/plugins/security_solution/public/common/components/guided_onboarding_tour/tour_step.tsx` , but wasn't able to access expandable context from there

Reply:
The flyout state is also synced in url state. You can probably check directly within `GuidedOnboardingTourStep`. For reference, there is an existing [hook](x-pack/plugins/security_solution/public/flyout/document_details/shared/hooks/use_is_timeline_flyout_open.ts) that checks whether timeline flyout is open.

Reply:
Agree, it's better to move this inside `SecurityTourStep`, with something like: 
```
const isHiddenByFlyout = useIsHiddenByFlyout(); 
```
that calls `const { search } = useLocation()` and checks the flyout state, if possible.